Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,402 @@
+        TRANSLATION_HIGHLIGHT_CLASS: 'is-selected',
+
+        events: {
+            '.ez-languageselectionbox-close': {
+                'tap': '_cancel',
+            },
+            '.ez-languageselectionbox-confirm': {
+                'tap': '_confirm',
+            },
+            '.ez-new-translation': {
+                'tap': '_selectNewTranslation'
+            },
+            '.ez-base-translation': {
+                'tap': '_selectBaseTranslation'
+            },
+        },
+
+        /**
+         * Tap event handler on the close links. It fires the `cancel` event.
+         *
+         * @method _cancel
+         * @protected
+         * @param {EventFacade} e
+         */
+        _cancel: function (e) {
+            e.preventDefault();
+            this.fire(CANCEL);
+        },
+
+        /**
+         * Tap event handler on the confirm button. It fires the `confirm` event.
+         *
+         * @method _confirm
+         * @protected
+         * @param {EventFacade} e
+         */
+        _confirm: function (e) {
+            e.preventDefault();
+            this.fire(CONFIRM);
+        },
+
+        /**
+         * Tap event handler on language from new translations list
+         *
+         * @method _selectNewTranslation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _selectNewTranslation: function (e) {
+            var languageCode = e.target.getAttribute('data-languagecode');
+
+            this.set('selectedNewTranslation', languageCode);
+        },
+
+        /**
+         * Tap event handler on language from existing translations list
+         *
+         * @method _selectBaseTranslation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _selectBaseTranslation: function (e) {
+            var languageCode = e.target.getAttribute('data-languagecode');
+
+            this.set('selectedBaseTranslation', languageCode);
+        },
+
+        /**
+         * Tap event handler on switcher dicatating if new translation will base
+         * on already existing one or not
+         *
+         * @method _baseTranslationToggle
+         * @protected
+         */
+        _baseTranslationToggle: function () {
+            this.set('baseTranslation', !this.get('baseTranslation'));
+        },
+
+        initializer: function () {
+            this.on(['confirmHandlerChange', 'cancelHandlerChange'], function (e) {
+                this._syncEventHandler(e.attrName.replace(/Handler$/, ''), e.prevVal, e.newVal);
+            });
+            this._syncEventHandler('confirm');
+            this._syncEventHandler('cancel');
+            this.after('activeChange', function () {
+                if ( this.get('active') ) {
+                    this._uiUpdateTitle();
+                }
+            });
+            this._publishEvents();
+            this.on(['existingTranslationsChange','canBaseTranslationChange'], function (e) {
```

Comment:
> 1. There are some view attributes used in template which may change. For example when state is being reset (when languageSelectionBox is closed) I render the view again. I wanted do the same, when attributes related to languages lists are changing: https://github.com/ezsystems/PlatformUIBundle/pull/280/files?diff=unified#diff-7add4a5b132be8c1d7fa15ac84ff9c8eR117
but it crashes - this line fails and return undefined instead of collection of available languages (currently set in LanguageSelectionBoxViewService).


for `canBaseTranslation` instead of trying to rerender completely the view when it changes, I would just add or remove a class on the container to show or hide the corresponding UI part.

for `existingTranslations`, this one should indeed trigger a render and I fail to see why it would not work.